Kudu.Core/SiteExtensions/FeedExtensions.cs

```diff
@@ -161,5 +162,20 @@ public static class FeedExtensions
                 }
             }
         }
+
+        private static async Task<T> GetResourceAndValidateAsync<T>(this SourceRepository srcRepo) where T : class, INuGetResource
+        {
```

Comment:
another way to do this is TryGetResourceAnd...().   And let the caller make decision.